cpp/pagerank_module/memgraph_utilities/mg_utility.cpp

```diff
@@ -0,0 +1,286 @@
+            edges.emplace_back(std::make_pair(first_node_map, second_node_map));
+            NE++;
+        }
+        // Destroy edge iterator
+        mgp_edges_iterator_destroy(edges_it);
+        NV++;
+    }
+    // Destroy vertices iterator
+    mgp_vertices_iterator_destroy(vertices_iterator);
+
+    auto *mapping = new GraphMapping;
+    mapping->numberOfVertices = NV;
+    mapping->numberOfEdges = NE;
+    mapping->edges = edges;
+    return mapping;
+}
+
+
+GraphMapping *MapMemgraphGraphWithPatterns(const mgp_list *nodes,
+                                           std::vector<std::vector<std::string>> patterns,
+                                           std::map<uint32_t, uint32_t> idToIter,
+                                           const mgp_graph *memgraphGraph,
+                                           mgp_memory *memory) {
+    long NV = 0, NE = 0;
+    // Iterating through vertices
+    std::vector<std::pair<uint32_t, uint32_t>> edges;
+    int nodes_size = mgp_list_size(nodes);
+    for (int i = 0; i < nodes_size; i++) {
+        const auto startVertex = mgp_value_get_vertex(mgp_list_at(nodes, i));
+
+        // For each vertex, iterate through its edges
+        const mgp_vertex *from = startVertex;
+        for (std::vector<std::string> pattern : patterns) {
+            std::set<int64_t> to_vertices = FindPattern(pattern, from, memgraphGraph, memory);
+            if (to_vertices.empty()) continue;
+
+            // Filter vertices that are available
+            auto it = to_vertices.begin();
+            while (it != to_vertices.end()) {
+                if (!idToIter.count(*it)) {
+                    it = to_vertices.erase(it);
+                } else ++it;
+            }
+
+            for (int64_t to_id: to_vertices) {
+                int64_t first_node_id = mgp_vertex_get_id(from).as_int;
+                int64_t second_node_id = to_id;
+
+                uint32_t first_node_map = idToIter[first_node_id];
+                uint32_t second_node_map = idToIter[second_node_id];
+
+                // Make list of edges
+                if (first_node_map > second_node_map) continue;
+                auto edge = std::make_pair(first_node_map, second_node_map);
+                if (std::count(edges.begin(), edges.end(), edge)) continue;
+                edges.emplace_back(edge);
+                NE++;
+            }
+        }
+        NV++;
+    }
+    auto *mapping = new GraphMapping;
+    mapping->numberOfVertices = NV;
+    mapping->numberOfEdges = NE;
+    mapping->edges = edges;
+    return mapping;
+}
+
+
+PairMap<uint32_t, uint32_t> VertexIdMappingSubgraph(const mgp_list *nodes,
+                                            const mgp_graph *graph,
+                                            mgp_result *result,
+                                            mgp_memory *memory) {
+    // Get vertices iterator
+    auto *vertices_it = GetVerticesIterator(graph, result, memory);
+
+    if (vertices_it == nullptr) {
```

Comment:
What's the point of the `GetVerticesIterator` if you do the same check like the one it's doing.